src/components/localization/language-switcher.js

```diff
@@ -17,19 +17,19 @@ class LanguageSwitch extends Component {
     componentWillReceiveProps(nextProps) {
         this.setState({ language: nextProps.i18n.language })
     }
-
-    renderLanguageChoice(lang) {
+    renderLanguageChoice = (lang) => {
         const { display_name, path } = language_config[lang]
         const to = `/${path}`
+        let text = this.props.shorthand === 'true' ? (display_name.substring(0, 3).toUpperCase()) : (display_name);
 
         return {
             value: to,
-            text: display_name,
+            text: text
         }
     }
     getCurrentLanguage() {
         const { display_name } = language_config[this.state.language]
-        return display_name
+        return this.props.shorthand === 'true' ? (display_name.substring(0, 3).toUpperCase()) : (display_name)
```

Comment:
i think this.state.language is the shorthand for the lang, can you check

Reply:
I think we should add `short_name` for languages in the config file to read from there.

Reply:
thats good idea 👍 , please proceed